### ISSUE TITLE
name resolution failed!  HELP!!!

### ISSUE THREAD

**Issue by gary170894** [aria2/aria2#774]
name resolution failed was show on every download HELP ...

**Comment by tatsuhiro-t** [comment]
Which OS are you running aria2 on?
Perhaps, --async-dns=false may workaround the issue.

**Comment by SoreGums** [comment]
Ubuntu 16.10.

aria2c downloaded a few files, then added a bridge adapter and rebooted, now aria2c doesn't download anymore...

> [AbstractCommand.cc:791] errorCode=19 CUID#7 - Name resolution for github-cloud.s3.amazonaws.com failed:no address returned or address conversion failed

**Comment by alexanderadam** [comment]
I have the same problem in Ubuntu 17.04. Maybe this could help: Ubuntu 17.04 is using systemd-resolved and addiontally installed dnsmasq.
The aria bug happened for me when I tried to installed to install a current ruby with [ruby-build](https://github.com/rbenv/ruby-build/blob/ae1f2850bb61552d6be16df11de4fd5dae94ad08/bin/ruby-build#L319-L326). `curl` and `wget` work out of the box and I can confirm that adding `--async-dns=false` to aria works as well.

**Comment by tatsuhiro-t** [comment]
I've heard this issue for quite some time, and still hasn't figured out the cause.  It may depend on c-ares version.  I suspect that it might not support advanced configurations in /etc/resolv.conf.

**Comment by alexanderadam** [comment]
@tatsuhiro-t is there anything I could do that could help or add some missing pieces of information?

**Comment by tatsuhiro-t** [comment]
@alexanderadam 
I have similar /etc/resolv.conf in my fully updated Ubuntu 17.04.
And aria2c works with libc-ares2 on it perfectly.
Do you have any special DNS configuration?

**Comment by tatsuhiro-t** [comment]
No.  My setup is pretty simple, and it only has dnsmasq-base, which merely just does DHCP stuff.
I'll test aria2c with dnsmasq installed.

**Comment by ikaros0327** [comment]
Hi @gary170894 i just have a same problem like you, and i'm on the windows 10 . i just edit my hosts file and it work.
the host file location on windows is "C:\Windows\System32\drivers\etc\hosts" 
and add your DNS result like 
113.43.202.238    31.gigafile.nu (use nslookup to change where you whant to download)

**Comment by Naville** [comment]
Can confirm this is still a bug on my Vanilla Ubuntu 20 in WSL2. Adding ``--async-dns=false`` bypasses this issue but still

**Comment by minyoad** [comment]
windows 10，Adding --async-dns=false worked

**Comment by hseg** [comment]
Using the stub resolver fixed the problem for me, though it is unclear why.
```
https://youtu.be/oVmDBGsvkYY

**Comment by foisonocean** [comment]
Same issue on Arch Linux:

![CleanShot 2021-01-26 at 16 02 07@2x](https://user-images.githubusercontent.com/15059605/105816978-f6f4e880-5fef-11eb-86e1-9f6e57468b48.png)

**Comment by Viech** [comment]
This happens to me on a fresh Arch install but not with one installed ~10 years ago. The relevant difference would be that I am now running `systemd-resolved`. Adding `--async-dns=false`works around the problem.

**Comment by ericisnotrealname** [comment]
I meet this issue on aria2 version 1.36.0
Compiler: gcc 11.1.0
  built by  x86_64-pc-linux-gnu
  on        Aug 21 2021 11:18:43
System: Linux 6.1.6-arch1-3 #1 SMP PREEMPT_DYNAMIC Mon, 16 Jan 2023 12:51:23 +0000 x86_64

**Comment by RobertKajnak** [comment]
I had the issue when using my local Japanese IP, but using a VPN to tunnel to the US worked...

**Comment by seisdr** [comment]
Ubuntu 23 
```
ubuntu@s484:~$ aria2c --async-dns=false   "magnet:?xt=urn:btih:b01a3a48980cf5053a7944e8ef2766c249ce15be&dn=rutor.info_God+of+War+%5Bv+1.0.1+%2B+OST%5D+%282022%29+PC+%7C+RePack+%D0%BE%D1%82+FitGirl&tr=udp://opentor.net:6969&tr=udp://opentor.org:2710&tr=http://retracker.local/announce"                                                                                                                                                            01/07 04:37:52 [NOTICE] Downloading 1 item(s)
                                                                                        01/07 04:37:52 [NOTICE] IPv4 DHT: listening on UDP port 6902
                                                                                        01/07 04:37:52 [NOTICE] IPv4 BitTorrent: listening on TCP port 6913                                                                                                             01/07 04:37:52 [NOTICE] IPv6 BitTorrent: listening on TCP port 6913                     [#409ada 0B/0B CN:0 SD:0 DL:0B]                                                         01/07 04:38:26 [ERROR] CUID#19 - Download aborted. URI=http://retracker.local/announce?info_hash=%B0%1A%3AH%98%0C%F5%05%3AyD%E8%EF%27f%C2I%CE%15%BE&peer_id=A2-1-36-0-%1D%E0%04%12%ADS%E0%EB%2F%9F&uploaded=0&downloaded=0&left=0&compact=1&key=%04%12%ADS%E0%EB%2F%9F&numwant=50&no_peer_id=1&port=6913&event=started&supportcrypto=1                          Exception: [AbstractCommand.cc:351] errorCode=19 URI=http://retracker.local/announce?info_hash=%B0%1A%3AH%98%0C%F5%05%3AyD%E8%EF%27f%C2I%CE%15%BE&peer_id=A2-1-36-0-%1D%E0%04%12%ADS%E0%EB%2F%9F&uploaded=0&downloaded=0&left=0&compact=1&key=%04%12%ADS%E0%EB%2F%9F&numwant=50&no_peer_id=1&port=6913&event=started&supportcrypto=1
  -> [NameResolver.cc:58] errorCode=19 Failed to resolve the hostname retracker.local, cause: Name or service not known
 *** Download Progress Summary as of Sun Jan  7 04:38:52 2024 ***                       ======================================================================================= [#409ada 0B/0B CN:0 SD:0 DL:0B]                                                         FILE: [MEMORY][METADATA]rutor.info_God+of+War+[v+1.0.1+++OST]+(2022)+PC+|+RePack+от+FitGirl                                                                                     ---------------------------------------------------------------------------------------                                                                                         [#409ada 0B/0B CN:0 SD:0 DL:0B]
```

**Comment by sbstn87** [comment]
I have the same issue on Windows 10. Adding --async-dns=false solves the issue, but I suppose that's not the way to handle this error in general.

At the same time when I use a virtual machine with Linux (Debian 12) aria2 works just fine without any additional parameters.

**Comment by zQueal** [comment]
Was having this issue with `yt-dlp` which offloaded download tasks to `aria2c`. Using;

> --downloader aria2c --downloader-args --async-dns=false

fixed this issue for me.

**Comment by Livy-Inverse** [comment]
After 10 years and it is still happening. aria2-1.37.0

**Comment by lokman2k5** [comment]
still having this issue, archlinux

**Comment by KSR-Yasuda** [comment]
`--async-dns=false` should be the default if the bug is sticking still, shouldn't it?
Happened on MSYS2 on Win11, aria2 v1.37.0.

**Comment by pingshunhuangalex** [comment]
Happens on Mac as well (`aria2 v1.37.0`). The homebrew version doesn't have the `--async-dns` feature. Not sure where to get a proper compiled version
